Add mocha

lmatiolis opened the pull request:
Para rodar os specs basta dar `grunt spec`.

Se quiser, pode abrir o arquivo `tests/nsn_build_test.html` no browser para ver os resultados mais organizados.

lmatiolis committed:
Add mocha, chai, es5-shim and grunt-mocha to setup test env.
Chai is being used as the assertion library for mocha.
es5-shim was necessary since phantomjs doesn't support bind(), which is used a lot in the project.

lmatiolis committed:
Add html to setup tests using build.js as the source.
Add small simple Engine test to show mocha working.

lmatiolis commented:
@claudiovicar code review

claudiovicar commented:
Bródi, por mim pode dar merge. Não consegui rodar aqui no Lemaf tb, o problema é no phantomjs então continua independente da lib de testes. Vou olhar com calma isso depois.

lmatiolis committed:
Add reporter option to mocha.

The 'Spec' reporter outputs a hierarchical view nested just as the test cases are.
The folder tests and specs where renamed too to follow mocha default pattern.

lmatiolis committed:
Separate tests in unit and integration.

//TODO
Build tests still not working because canvas is not being rendered.
The problem is with the assets loading...

lmatiolis committed:
Add Sinon and Sinon-Chai to use mocks and stubs.

claudiovicar committed:
Prefixing the file paths to make tests work

claudiovicar committed:
Fixing the first integration test - calling init() only once

lmatiolis committed:
Stub GameSound in Engine unit test.

Adds sinon dependencies to allow stub.

lmatiolis committed:
Fix call to mocha.run in engine.test.html

lmatiolis committed:
Fix init_game.spec description.

It was describing the Engine.

lmatiolis committed:
Sets window.nsn in the tests' html.

claudiovicar committed:
Fixing filter path
